### ISSUE TITLE
When using STDIN, updates to source code aren't recognised

### ISSUE THREAD

**Issue by Geraint** [phpmd/phpmd#941]
- PHPMD version: 2.11.1
- PHP Version: 7.4.27, 8.0.16, 8.1.1
- Installation type: composer
- Operating System / Distribution & Version: Windows 10/WSL2 running Ubuntu 20.04

### Current Behavior

Given this file, `test.php`:

```php
<?php

class Foo
{
    public function bar()
    {
        $a = 9;
    }
}
```

... I run this:

```bash
cat test.php | ./vendor/bin/phpmd - text cleancode,codesize,controversial,design,naming,unusedcode
```

... and get this (expected) output:

```
php://stdin:7  ShortVariable        Avoid variables with short names like $a. Configured minimum length is 3.
php://stdin:7  UnusedLocalVariable  Avoid unused local variables such as '$a'.
```

This is the same output as when I run `phpmd` using the filename rather than STDIN:

```bash
./vendor/bin/phpmd test.php text cleancode,codesize,controversial,design,naming,unusedcode
```

```
/home/geraint/workspace/sandbox/phpmd-test/test.php:7  ShortVariable        Avoid variables with short names like $a. Configured minimum length is 3.
/home/geraint/workspace/sandbox/phpmd-test/test.php:7  UnusedLocalVariable  Avoid unused local variables such as '$a'.
```

However, if I change the variable `$a` to `$b` such that `test.php` now looks like this:

```php
<?php

class Foo
{
    public function bar()
    {
        $b = 9;
    }
}
```

... and run `phpmd` again:

```bash
cat test.php | ./vendor/bin/phpmd - text cleancode,codesize,controversial,design,naming,unusedcode
```

... I (unexpectedly) get the same results as before (`$a` is still reported as the problematic variable, not `$b`):

```
php://stdin:7  ShortVariable        Avoid variables with short names like $a. Configured minimum length is 3.
php://stdin:7  UnusedLocalVariable  Avoid unused local variables such as '$a'.
```

If, in contrast, I use the filename, rather than STDIN, I get expected output:

```bash
./vendor/bin/phpmd test.php text cleancode,codesize,controversial,design,naming,unusedcode
```

```
/home/geraint/workspace/sandbox/phpmd-test/test.php:7  ShortVariable        Avoid variables with short names like $b. Configured minimum length is 3.
/home/geraint/workspace/sandbox/phpmd-test/test.php:7  UnusedLocalVariable  Avoid unused local variables such as '$b'.
```

If I clear out the `.pdepend` directory:

```bash
rm -rf ~/.pdepend
```

... and run `phpmd` using STDIN again:

```bash
cat test.php | ./vendor/bin/phpmd - text cleancode,codesize,controversial,design,naming,unusedcode
```

... then the output is as expected (i.e. updates to the source code are reflected in the output):

```
php://stdin:7  ShortVariable        Avoid variables with short names like $b. Configured minimum length is 3.
php://stdin:7  UnusedLocalVariable  Avoid unused local variables such as '$b'.
```

A further oddity I have noticed is that, having run the command using standard input at least once, I can run it again, without even piping anything into it:

```bash
./vendor/bin/phpmd - text cleancode,codesize,controversial,design,naming,unusedcode
```

... and still get output:

```
php://stdin:7  ShortVariable        Avoid variables with short names like $b. Configured minimum length is 3.
php://stdin:7  UnusedLocalVariable  Avoid unused local variables such as '$b'.
```

In contrast, if I clear the `.pdepend` directory before I run `phpmd`:

```bash
rm -rf ~/.pdepend
./vendor/bin/phpmd - text cleancode,codesize,controversial,design,naming,unusedcode
```

... then the command just stops, waiting for input (as expected).

It seems STDIN might somehow be getting cached in the `.pdepend` directory.

I have tried this with PHP 7.4, 8.0 and 8.1, all yielding the same results.

I have also observed the same behaviour when I use JSON output.

### Expected Behavior

I expect the output to reflect updates to source code without having to manually clear the `.pdepend` directory when I use STDIN, as it does when I used the filename.

### Steps To Reproduce:

Installed phpmd by creating this `composer.json`:

```json
{
    "require-dev": {
        "phpmd/phpmd" : "@stable"
    }
}
```

... and then running

```bash
composer install
```

Created `test.php`, as above.

Ran `phpmd`, using STDIN:

```bash
cat test.php | ./vendor/bin/phpmd - text cleancode,codesize,controversial,design,naming,unusedcode
```

Updated `test.php` and ran `phpmd` again, and observed that output did not reflect the newly made update.

## Checks before submitting
* [X] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [X] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [X] I have added every step to reproduce the bug.
* [X] If possible I added relevant code examples.
* [X] This issue is about 1 bug and nothing more.
* [X] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by kylekatarnls** [comment]
🙏 Thanks for this well detailed bug report. Indeed, the hash for the cache in this case is `md5('php://stdin')` which is constant :(

I fixed it to use the source code instead.

You can check it by `composer require "pdepend/pdepend:dev-fix/cache-stdin as 2.10.4"`

**Comment by Geraint** [comment]
@kylekatarnls 

> You can check it by `composer require "pdepend/pdepend:dev-fix/cache-stdin as 2.10.4"`

This fixes it for me.  Many thanks.
